src/cpp/util/byte_buffer.cc

```diff
@@ -95,4 +102,10 @@ ByteBuffer& ByteBuffer::operator=(const ByteBuffer& buf) {
   return *this;
 }
 
+void ByteBuffer::Swap(ByteBuffer* other) {
+  grpc_byte_buffer* tmp = other->buffer_;
+  other->buffer_ = this->buffer_;
```

Comment:
For the record, the reason I used "this" was to contrast with "other".

When a method is accessing the fields of multiple instances, I find it
helps readability to make the instance explicit. Especially so when the
instances are used symmetrically, as they
are here:  that is,  x.Swap(y)  is equivalent to y.Swap(x).

But of course, you should do whatever you feel makes it easier for you to
understand.

Thanks for taking the change.

On 18 August 2016 at 10:05, Vijay Pai <notifications@github.com> wrote:

> In src/cpp/util/byte_buffer.cc
> <https://github.com/grpc/grpc/pull/7773#discussion_r75347450>:
>
> > @@ -95,4 +102,10 @@ ByteBuffer& ByteBuffer::operator=(const ByteBuffer& buf) {
> >    return *this;
> >  }
> >
> > +void ByteBuffer::Swap(ByteBuffer* other) {
> > +  grpc_byte_buffer* tmp = other->buffer_;
> > +  other->buffer_ = this->buffer_;
>
> Yeah, I like an explanation wherever I see "this". There is at least 1
> case where it's needed: template disambiguation. Aside from that, I tend to
> think it's confusing. Will send a follow-on PR.
> … <#m_7421317622789186882_>
> On Thu, Aug 18, 2016 at 9:53 AM Mark D. Roth ***@***.***> wrote: In
> src/cpp/util/byte_buffer.cc <#7773 (comment)
> <https://github.com/grpc/grpc/pull/7773#discussion_r75345611>>: > @@
> -95,4 +102,10 @@ ByteBuffer& ByteBuffer::operator=(const ByteBuffer& buf) {
> > return *this; > } > > +void ByteBuffer::Swap(ByteBuffer* other) { > +
> grpc_byte_buffer* tmp = other->buffer_; > + other->buffer_ = this->buffer_;
> Yeah, that shouldn't be necessary. I copied that from the internal CL and
> didn't notice it before I merged. Feel free to send me a PR to fix it if it
> bothers you. — You are receiving this because you commented. Reply to this
> email directly, view it on GitHub <https://github.com/grpc/grpc/
> pull/7773/files/2665bdd6d29ec2515946ac629002c04c23a50bfa#r75345611>, or
> mute the thread <https://github.com/notifications/unsubscribe-
> auth/AIdEkJjQ_u_jjjlxxCjykq9s5ni2gIUZks5qhI4VgaJpZM4Jnln5> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc/pull/7773/files/2665bdd6d29ec2515946ac629002c04c23a50bfa#r75347450>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJsp2RswEortmBMhcnJUiU02vgfYJT5bks5qhJDRgaJpZM4Jnln5>
> .
>